src/server/game/Entities/GameObject/GameObject.cpp

```diff
@@ -2777,6 +2777,16 @@ class GameObjectModelOwnerImpl : public GameObjectModelOwnerBase
     GameObject* _owner;
 };
 
+void GameObject::SetSpellVisual(int32 spellVisualId, ObjectGuid activatorGuid)
+{
+    SetUpdateFieldValue(m_values.ModifyValue(&GameObject::m_gameObjectData).ModifyValue(&UF::GameObjectData::SpellVisualID), spellVisualId);
```

Comment:
should it really always set the visual permanently?

Reply:
I only checked the capture points in Arathi Basin and Battle for Gilneas at the end of the legion expansion, the value was send in the update object packet after. Not sure if other types of gameobject have different behaviour